src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java

```diff
@@ -131,6 +133,67 @@ public Mono<XPayAuthResponseEntityDto> requestXPayAuthorization(AuthorizationReq
     }
 
+    public Mono<CreditCardAuthResponseDto> requestCreditCardAuthorization(AuthorizationRequestData authorizationData) {
+        return Mono.just(authorizationData)
+                .filter(
+                        authorizationRequestData -> "CP".equals(authorizationRequestData.paymentTypeCode())
+                                && "VPOS".equals(authorizationRequestData.paymentGatewayId())
+                )
+                .switchIfEmpty(Mono.empty())
+                .flatMap(authorizationRequestData -> {
+                    final Mono<CreditCardAuthRequestDto> creditCardAuthRequest;
+                    if (authorizationData.authDetails()instanceof CardAuthRequestDetailsDto cardData) {
+                        BigDecimal grandTotal = BigDecimal.valueOf(
+                                ((long) authorizationData.transaction().getPaymentNotices().stream()
+                                        .mapToInt(paymentNotice -> paymentNotice.transactionAmount().value()).sum())
+                                        + authorizationData.fee()
+                        );
+                        creditCardAuthRequest = Mono.just(
+                                new CreditCardAuthRequestDto()
+                                        .pan(cardData.getPan())
+                                        .expireDate(cardData.getExpiryDate())
+                                        .idTransaction(
+                                                authorizationData.transaction().getTransactionId().value().toString()
+                                        )
+                                        .amount(grandTotal)
+                                        .emailCh(authorizationData.transaction().getEmail().value())
+                                        .holder(cardData.getHolderName())
+                                        .securitycode(cardData.getCvv())
+                                        .isFirstPayment(true) // TODO TO BE CHECKED
+                                        .threeDsData("threeDsData") // TODO From client
+                                        .circuit("VISA") // TODO Add field to card details
+                                        .reqRefNumber("reqRefNumber") // TODO From client
+                                        .idPsp(authorizationData.pspId())
+                        );
+                    } else {
+                        creditCardAuthRequest = Mono.error(
+                                new InvalidRequestException(
+                                        "Cannot perform VPOS authorization for null input CreditCardAuthRequestDto"
+                                )
+                        );
+                    }
+                    return creditCardAuthRequest;
+                })
+                .flatMap(
+                        creditCardAuthRequestDto -> creditCardInternalApiClient
+                                .step0CreditCard(
+                                        "Web",
```

Comment:
the clientId does not need to be set, but is generated by the apim based on the subscription key. If the autogenerated code requires it, you need to fix the pgs openapi.
https://raw.githubusercontent.com/pagopa/pagopa-infra/main/src/core/api/payment_transactions_gateway_api/internal/v1/_openapi.json.tpl